sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs

```diff
@@ -408,35 +398,39 @@ public DataLakeFileSystemClient(Uri fileSystemUri, TokenCredential credential, D
         /// A <see cref="Uri"/> referencing the file system that includes the
         /// name of the account and the name of the file system.
         /// </param>
-        /// <param name="pipeline">
-        /// The transport pipeline used to send every request.
-        /// </param>
-        /// <param name="storageSharedKeyCredential">
-        /// The shared key credential used to sign requests.
-        /// </param>
-        /// <param name="version">
-        /// The version of the service to use when sending requests.
-        /// </param>
-        /// <param name="clientDiagnostics">
-        /// The <see cref="ClientDiagnostics"/> instance used to create
-        /// diagnostic scopes every request.
+        /// <param name="clientConfiguration">
+        /// <see cref="DataLakeClientConfiguration"/>.
         /// </param>
         internal DataLakeFileSystemClient(
             Uri fileSystemUri,
-            HttpPipeline pipeline,
-            StorageSharedKeyCredential storageSharedKeyCredential,
-            DataLakeClientOptions.ServiceVersion version,
-            ClientDiagnostics clientDiagnostics)
+            DataLakeClientConfiguration clientConfiguration)
         {
             DataLakeUriBuilder uriBuilder = new DataLakeUriBuilder(fileSystemUri);
             _uri = fileSystemUri;
             _blobUri = uriBuilder.ToBlobUri();
             _dfsUri = uriBuilder.ToDfsUri();
-            _pipeline = pipeline;
-            _storageSharedKeyCredential = storageSharedKeyCredential;
-            _version = version;
-            _clientDiagnostics = clientDiagnostics;
-            _containerClient = BlobContainerClientInternals.Create(_blobUri, pipeline, _version.AsBlobsVersion(), _clientDiagnostics);
+
+            _clientConfiguration = clientConfiguration;
+
+            _containerClient = BlobContainerClientInternals.Create(
+                _blobUri,
+                _clientConfiguration);
+
+            _fileSystemRestClient = BuildFileSystemRestClient(fileSystemUri);
+        }
+
+        private FileSystemRestClient BuildFileSystemRestClient(Uri uri)
+        {
+            DataLakeUriBuilder uriBuilder = new DataLakeUriBuilder(uri);
+            string fileSystemName = uriBuilder.FileSystemName;
+            uriBuilder.FileSystemName = null;
+            // TODO get rid of resource parameter on FileSystemRestClient constructor.
```

Comment:
bubbling this up to not forget.

Reply:
Yep this is the last TODO I need to address.  There's going to be some serious swagger readme transform to get rid of that parameter.